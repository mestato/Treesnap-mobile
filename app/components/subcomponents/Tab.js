import React, {Component, PropTypes} from 'react'
import {View, StyleSheet, Dimensions} from 'react-native'

export default class Tab extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <View style={[styles.container, {width: Dimensions.get('window').width}]}>
        {this.props.children}
      </View>
    )
  }

}

Tab.PropTypes = {
  title: PropTypes.string.isRequired
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  }
})