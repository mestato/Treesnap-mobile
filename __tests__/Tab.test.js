import React from 'react';
import Tab from '../app/components/subcomponents/Tab'
import renderer from 'react-test-renderer'

describe('it renders correctly', () => {
  it('should match a tab snapshot', () => {
    const tree = renderer.create(<Tab title={'Test Title'}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
