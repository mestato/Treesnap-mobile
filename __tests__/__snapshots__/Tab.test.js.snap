// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly should match a tab snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "width": 750,
      },
    ]
  }
/>
`;
